### DISCUSSION TITLE
Cannot Activate/SwitchTo "ActiveWindow"

### DISCUSSION THREAD

**Discussion by GHRyunosuke** [GCuser99/SeleniumVBA#113] [Q&A] [answered]
Here is how to reproduce this issue:

1. Use  `driver.NavigateTo strURL` to go to some website --> I suppose the tab that is now showing the web content is the "ActiveWindow"
2. Open a new tab manually, so you now switch to another tab.
3. Say you want to switch back/activate the original tab (i.e. the "ActiveWindow" above). I tried the following 2 pieces of code, and none of those works: `driver.ActiveWindow.SwitchTo` or `driver.ActiveWindow.Activate` (both could be executed, but nothing happens...)
4. The only code that works is `driver.Windows.SwitchToByUrl driver.ActiveWindow.Url` which is a little awkward to switch to the already-known active window. 

 `driver.ActiveWindow.SwitchTo` or `driver.ActiveWindow.Activate` seems to be more straightforward?

**Comment by GCuser99** [comment, marked as answer]
Hi @GHRyunosuke,

Yeah, the Selenium WebDriver does not recognize a tab that is created manually as the active window (although it still aware of the newly created window) - I would call it a limitation as opposed to a bug. So the first (non-manually created) window is still the active window as far as the WebDriver is concerned. An alternative to using `driver.Windows.SwitchToByUrl` and `driver.ActiveWindow.Url` is the following:
```vba
Dim driver As WebDriver
Dim oldWindow As WebWindow

Set driver = New WebDriver

driver.StartChrome
driver.OpenBrowser

driver.NavigateTo "http://the-internet.herokuapp.com/windows"

Set oldWindow = driver.ActiveWindow

Stop 'create a new tab manually

driver.Windows.SwitchToNext 'make the new window the active one
oldWindow.SwitchTo 'now switch back to the old one

'also following will work too if you only have two open windows
'driver.Windows.SwitchToNext 'make the new window the active one
'driver.Windows.SwitchToNext 'now switch back to the old one

driver.Wait 3000

driver.Shutdown
```

**Reply by GHRyunosuke** [reply]
Interesting... thank you & got it.
This alternative is better than "SwitchToByUrl" and "ActiveWindow.Url", will use this instead.
